src/application.js

```diff
@@ -127,6 +127,8 @@ export default class Application extends Adw.Application {
         if (!this.window) {
             this.window = new Window({ application: this });
             this.window.hide_on_close = settings.get_boolean('hide-on-close');
+        } else {
+            this.window.present();
```

Comment:
Actually, this should replace the `this.window.show();` in the conditional bellow. Otherwise it will break the "Start Hidden" preference.

Reply:
Okay, I moved it.